### DISCUSSION TITLE
Cannot use `add_head_html` or `ui.colors()` in `single_page_app` example!

### DISCUSSION THREAD

**Discussion by KiddoV** [zauberzeug/nicegui#2630] [Q&A] [answered]
### Question

How can I add these methods before ``ui.run()`` in this example: https://github.com/zauberzeug/nicegui/tree/310453bd6cee63c32887876c4471b189efbec094/examples/single_page_app
```py
app.add_static_files()

ui.add_head_html()
ui.colors(...)
```
I have tried those, but didn't work!

**Comment by falkoschindler** [comment, marked as answer]
Hi @KiddoV,
Did you put `ui.add_head_html()` and `ui.colors()` into the `main()` function? Since this function creates a new route, UI elements from the global scope won't show up there.

**Reply by KiddoV** [reply]
That works, thanks!

I have to take the ``app.add_static_files()`` outside of ``main()`` and put ``ui.add_head_html()`` (or anything with ``ui``, except ``ui.run()``) inside it. When I put the ``app.add_static_files()`` inside, it doesn't work for some reason.

**Comment by KiddoV** [comment]
Also, I have another question.
How do I make the [single page app example](https://github.com/zauberzeug/nicegui/tree/310453bd6cee63c32887876c4471b189efbec094/examples/single_page_app) buttons change color when its route is active, like a tab behavior?

**Reply by falkoschindler** [reply]
You can assign the buttons to variables `b1`, `b2`, `b3` and call something like `b1.props('color=red')` inside the `show_` functions.
